### PR TITLE
feat: Realtime update for theme buttons

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description

Please include a summary of the change and issue(s) fixed. Also, mention relevant motivation, context, and any dependencies that this change requires.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/javascript/portal/portalThemeHelper.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
+    '--dynamic-hover-color',
+    hoverColor
+  );
+};
+
+export const setPortalClass = theme => {
+  const portalDiv = document.querySelector('#portal');
```
Comment (scmmishra) [inline app/javascript/portal/portalThemeHelper.js:21]:
```suggestion
  const portalDiv = document.getElementById('portal');
```

File: app/javascript/portal/portalThemeHelper.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+export const updateThemeStyles = theme => {
+  setPortalClass(theme);
+  setPortalHoverColor(theme);
+};
+
+export const toggleAppearanceDropdown = () => {
+  const dropdown = document.getElementById('appearance-dropdown');
+  if (!dropdown) return;
+  dropdown.style.display =
```
Comment (scmmishra) [inline app/javascript/portal/portalThemeHelper.js:35]: Let's simplify this line. It's not very readable

**Review by scmmishra — CHANGES_REQUESTED** [review]

File: app/javascript/portal/portalThemeHelper.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+const SELECTED_COOKIE = 'selected_theme';
+
+export const setPortalHoverColor = theme => {
+  if (window.portalConfig.isPlainLayoutEnabled === 'true') return;
+  const portalColor = window.portalConfig.portalColor;
+  const bgColor = theme === 'dark' ? '#151718' : 'white';
+  const hoverColor = adjustColorForContrast(portalColor, bgColor);
+
+  // Set hover color for border and text dynamically
+  document.documentElement.style.setProperty(
```
Comment (scmmishra) [inline app/javascript/portal/portalThemeHelper.js:14]: Do we need to do this in JS? is it not possible in the ERB template?

Comment (iamsivin) [reply app/javascript/portal/portalThemeHelper.js:14]: @scmmishra I think we need to handle it from here because we are setting `hoverColor` from color2k.

Comment (scmmishra) [reply app/javascript/portal/portalThemeHelper.js:14]: Noted, we can move the implementation to backend anyway, I'm sure there will be a `color2k` like library there. But can be handled later

Comment (iamsivin) [reply app/javascript/portal/portalThemeHelper.js:14]: Yeah, I will find out a good gem and use it.

**Pull request merged**